ligo/src/types.mligo

```diff
@@ -0,0 +1,230 @@
+// Corresponds to Types.hs module
+
+// -- FA2 types -- //
+
+#if !TYPES_H
+#define TYPES_H
+
+type token_id = nat
+type token_metadata =
+  [@layout:comb]
+  { token_id : token_id
+  ; symbol : string
+  ; name : string
+  ; decimals : nat
+  ; extras : (string, string) map
+  }
+
+type operator =
+  { owner : address
+  ; operator : address
+  }
+type operators = (operator, unit) big_map
+
+type ledger_key = address * token_id
+type ledger_value = nat
+type ledger = (ledger_key, ledger_value) big_map
+
+// -- DAO base types -- //
+
```

Comment:
I see that there is no equivalent for the `Nonce` type, I suppose that's because it's less useful without the options to make `newtype`s in LIGO, but I'd argue even just a `nat` type synonym could be nice to have (as we do for the other ones).

Reply:
Okay, added the necessary types to make Lorentz and LIGO versions match.

I only skip `DataToSign`, because IMO there is no sense in a non-polymorphic version of this type; I'll let the implementor of signing logic decide whether to use a record here or just pairs.